pyqtgraph/functions.py

```diff
@@ -1111,7 +1111,7 @@ def makeARGB(data, lut=None, levels=None, scale=None, useRGBA=False):
     # awkward, but fastest numpy native nan evaluation
     # 
     nanMask = None
-    if data.dtype.kind == 'f' and np.isnan(data.min()):
+    if data.ndim == 2 and data.dtype.kind == 'f' and np.isnan(data.min()):
```

Comment:
For a given pixel where one or more of the color channel values is nan, the color of that pixel is kind of "undefined", so making it transparent makes the most sense I think. We could collapse the nan mask along the color channel axis and still use it:

```suggestion
    if data.dtype.kind == 'f' and np.isnan(data.min()):
        nanMask = np.isnan(data)
        if data.ndim > 2:
            nanMask = np.any(nanMask, axis=data.ndim-1)
```